src/transfersavesdialog.cpp

```diff
@@ -285,34 +217,13 @@ void TransferSavesDialog::refreshSaves(SaveCollection &saveCollection, QString c
 {
   saveCollection.clear();
 
-  SaveGameInfo const *info = m_GamePlugin->feature<SaveGameInfo>();
-  if (info == nullptr) {
-    static DummyInfo dummyInfo;
-    info = &dummyInfo;
-  }
-
-  QStringList filters;
-  filters << QString("*.") + m_GamePlugin->savegameExtension();
-
-  QDir savesDir(savedir);
-  savesDir.setNameFilters(QStringList() << QString("*.") + m_GamePlugin->savegameExtension());
-  savesDir.setFilter(QDir::Files);
-  QDirIterator it(savesDir, QDirIterator::Subdirectories);
-  log::debug("reading save games from {}", savesDir.absolutePath());
-
-  QFileInfoList files;
-  while (it.hasNext()) {
-    it.next();
-    files.append(it.fileInfo());
-  }
-  std::sort(files.begin(), files.end(), [](auto const& lhs, auto const& rhs) {
-    return lhs.fileTime(QFileDevice::FileModificationTime) < rhs.fileTime(QFileDevice::FileModificationTime);
+  auto saves = m_GamePlugin->listSaves(savedir);

```

Comment:
What happens here if the game doesn't implement `listSaves()`?

Reply:
This is not a feature so the game always implement `listSaves`. If the game return an empty list, the list will be empty.

Reply:
It's not the what I would have thought knowing there is a game feature specific for saves. But I guess that is for the hover widget only?

Reply:
The game feature is for widgets and missing assets.